rsdockerspawner/rsdockerspawner.py

```diff
@@ -308,7 +309,7 @@ def __init_containers(cls, pool, log):
                                 s.num,
                                 s.host,
                             )
-                            cls.__slot_assign(s, n)
+                            cls.__slot_assign(s, n, previous_slot=previous_slots[n])
```

Comment:
What if there is no previous slot? This could happen if the slot file gets deleted or manually edited.

Reply:
Can there be no previous slot? I don't fully understand the code so please lmk but from my reading of it when containers are garbage collected they are removed. So, the only running containers should be in previous_slots (assuming the file is kept up to date...).

Reply:
There's no guarantee that the container has a slot. It could be the user got moved to the pool. If you change the `[n]` to a `get(n)`, it would be better. The file might not exist, for example. Imagine we want to reset the state so we remove the file, and then restart the server.

Reply:
1ef7bf1

> There's no guarantee that the container has a slot
I hope I'm not being annoying but I'd like to understand this code better. What is the scenario where this happens? My thinking with not using a get was that it was "fail fast". If we have running containers that we didn't previously know about then we have a bug and it'd be nice to know.

Reply:
When users used to be banned, we can have containers that could not possibly exist. What we would have to do is remove the containers by hand. rsdockerspawner should kill any containers it finds with port labels that it doesn't understand.